packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -159,86 +163,103 @@ const Trade = observer(() => {
-                        <SwipeableWrapper
-                            onChange={onChangeSwipeableIndex}
-                            is_disabled={
-                                !show_digits_stats ||
-                                !is_trade_enabled ||
-                                form_components.length === 0 ||
-                                is_chart_loading ||
-                                should_show_active_symbols_loading
-                            }
-                            is_swipe_disabled={swipe_index === 1}
-                            className={classNames({ 'vanilla-trade-chart': is_vanilla })}
-                        >
-                            {show_digits_stats && <DigitsWidget digits={digits} tick={tick} />}
-                            <ChartTrade
-                                topWidgets={topWidgets}
-                                charts_ref={charts_ref}
-                                bottomWidgets={show_digits_stats ? bottomWidgets : undefined}
-                                is_accumulator={is_accumulator}
-                            />
-                        </SwipeableWrapper>
-                    </MobileWrapper>
-                </React.Suspense>
-
-                {/* Remove Test component for debugging below for production release */}
-                <Test />
-            </Div100vhContainer>
-            <div className={form_wrapper_class}>
-                {is_market_closed && !is_market_unavailable_visible && (
-                    <MarketIsClosedOverlay
-                        is_eu={is_eu}
-                        is_synthetics_trading_market_available={is_synthetics_trading_market_available}
-                        {...(is_eu && category)}
-                        onClick={onTryOtherMarkets}
-                        onMarketOpen={prepareTradeStore}
-                        symbol={symbol}
+                    <NotificationMessages />
+                    <React.Suspense
+                        fallback={<ChartLoader is_dark={is_dark_theme} is_visible={!symbol || is_chart_loading} />}
+                    >
+                        <DesktopWrapper>
+                            <div
+                                className={classNames('chart-container__wrapper', {
+                                    'vanilla-trade-chart': is_vanilla,
+                                })}
+                            >
+                                <ChartLoader is_visible={is_chart_loading || should_show_active_symbols_loading} />
+                                <ChartTrade
+                                    topWidgets={topWidgets}
+                                    charts_ref={charts_ref}
+                                    is_accumulator={is_accumulator}
+                                />
+                            </div>
+                        </DesktopWrapper>
+                        {!(openLaunchModal && is_logged_in && !is_already_shown) && (
```

Comment:
It is the opposite statement of already written one on line 173:
```
{openLaunchModal && is_logged_in && !is_already_shown && (
                <LaunchModal handleChange={handleLaunchModal} open={openLaunchModal} />
)}
```
  Maybe it can be somehow refactored with the usage of ternary and etc?

Reply:
Actually both of them are in different heirarchy 

```
{openLaunchModal && is_logged_in && !is_already_shown && (
                <LaunchModal handleChange={handleLaunchModal} open={openLaunchModal} />
)}
```
```
<div>
<Div100vhContainer>
                        {!(open_launch_modal && is_logged_in && !is_already_shown) && (
                                                    <MobileWrapper>
                                <ChartLoader is_visible={is_chart_loading || should_show_active_symbols_loading} />
                                <SwipeableWrapper
                                    onChange={onChangeSwipeableIndex}
                                    is_disabled={
                                        !show_digits_stats ||
                                        !is_trade_enabled ||
                                        form_components.length ||
                                        is_chart_loading ||
                                        should_show_active_symbols_loading
                                    }
                                    is_swipe_disabled={swipe_index === 1}
                                    className={classNames({ 'vanilla-trade-chart': is_vanilla })}
                                >
                                    {show_digits_stats && <DigitsWidget digits={digits} tick={tick} />}
                                    <ChartTrade
                                        topWidgets={topWidgets}
                                        charts_ref={charts_ref}
                                        bottomWidgets={show_digits_stats ? bottomWidgets : undefined}
                                        is_accumulator={is_accumulator}
                                    />
                                </SwipeableWrapper>
                            </MobileWrapper>)
</Div100vhContainer>
</div>
```

I am not sure if we can do that

Reply:
Sure, no worries, thank you for checking